addon-test-support/@ember/test-helpers/dom/blur.ts

```diff
@@ -24,14 +29,17 @@ export function __blur__(element: HTMLElement | Element | Document | SVGElement)
 
   // makes `document.activeElement` be `body`.
   // If the browser is focused, it also fires a blur event
-  element.blur();
+  if (!relatedTarget) {
+    element.blur();
+  }
 
   // Chrome/Firefox does not trigger the `blur` event if the window
   // does not have focus. If the document does not have focus then
   // fire `blur` event via native event.
   if (browserIsNotFocused) {
-    fireEvent(element, 'blur', { bubbles: false });
-    fireEvent(element, 'focusout');
+    let options = { relatedTarget };
+    fireEvent(element, 'blur', assign({ bubbles: false }, options));
+    fireEvent(element, 'focusout', options);
   }
```

Comment:
I'm thinking we need something like:

```ts
  let browserIsNotFocused = document.hasFocus && !document.hasFocus();
  let needsCustomEventOptions = relatedTarget !== null;

  if (!needsCustomEventOptions) {
    // makes `document.activeElement` be `body`.
    // If the browser is focused, it also fires a blur event
    element.blur();
  }

  // Chrome/Firefox does not trigger the `blur` event if the window
  // does not have focus. If the document does not have focus then
  // fire `blur` event via native event.
  if (browserIsNotFocused || needsCustomEventOption) {
    let options = { relatedTarget };
    fireEvent(element, 'blur', assign({ bubbles: false }, options));
    fireEvent(element, 'focusout', options);
  }
```

Sorry, it wouldn't let me make a suggestion for this.

Reply:
I've incorporated this snippet. Only minor change is that we don't need the extra ` || needsCustomEventOption` check, as `fireEvent` should only execute when browser is not focused.